packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx

```diff
@@ -1014,6 +1016,30 @@ const UsernamePasswordAuthenticationStrategyEditor = observer(
     );
   },
 );
+// Middle Tier Username Password Authentication Strategy obtains credentials(username and password) from the provided vault reference
+const MiddleTierUsernamePasswordAuthenticationStrategyEditor = observer(
+  (props: {
+    authSpec: MiddleTierUsernamePasswordAuthenticationStrategy;
+    isReadOnly: boolean;
+  }) => {
+    const { authSpec, isReadOnly } = props;
+    return (
+      <>
+        <ConnectionEditor_StringEditor
+          isReadOnly={isReadOnly}
+          value={authSpec.vaultReference}
+          propertyName={'vault reference'}
+          update={(value: string | undefined): void =>
```

Comment:
should we provide a description for this component so the user knows what this property is for?

Reply:
Added in this [commit](https://github.com/finos/legend-studio/pull/1238/commits/33190594ae714703d426ac7198aebfa73fed762d)